neutrino.go

```diff
@@ -898,6 +901,79 @@ func (s *ChainService) NotifyMempoolReceived(addrs []bchutil.Address) {
 	s.mempool.NotifyReceived(addrs)
 }
 
+// RequestMempoolFilter requests the mempool filter from a single peer.
+func (s *ChainService) RequestMempoolFilter(addrs []bchutil.Address) {
+	go s.requestMempoolFilter(addrs)
+}
+
+func (s *ChainService) requestMempoolFilter(addrs []bchutil.Address) {
+	// We'll just start with our first peer and if that one fails
+	// move on to the next one. Since the mempool filter is not
+	// authenticated against a block, the remote peer could lie
+	// to us and omit transactions but we're treatign this function
+	// as best effort anyway. Worst case scenario we either download
+	// the mempool unnecessarily or have to wait for confirmation to
+	// detect our transaction.
+	for _, peer := range s.Peers() {
+		msgChan := make(chan spMsg)
+		subQuit := make(chan struct{})
+		subscription := spMsgSubscription{
+			msgChan:  msgChan,
+			quitChan: subQuit,
+		}
+		defer close(subQuit)
+
+		// Subscribe to the response
+		peer.subscribeRecvMsg(subscription)
+		peer.Peer.QueueMessage(wire.NewMsgGetCFMempool(), nil)
+
+		timeout := time.After(QueryPeerConnectTimeout)
+		select {
+		case <-timeout:
+			peer.unsubscribeRecvMsgs(subscription)
+			continue
+		case msg := <-msgChan:
```

Comment:
Messages other than the response could come over the wire here. We need a loop here where we check the message type until the timeout.